examples/vault-iam-auth/main.tf

```diff
@@ -0,0 +1,285 @@
+    "${aws_security_group.auth_instance.id}",
+  ]
+
+  user_data            = "${data.template_file.user_data_auth_client.rendered}"
+  iam_instance_profile = "${aws_iam_instance_profile.example_instance_profile.name}"
+
+  tags {
+    Name = "${var.auth_server_name}"
+  }
+
+  # These two blocks below exist to provision a script that uses an AWS SDK to sign a request
+  provisioner "file" {
+    source      = "${path.module}/auth-signature-scripts"
+    destination = "/tmp"
+  }
+
+  connection {
+    type        = "ssh"
+    user        = "${var.ssh_username}"
+    private_key = "${file("${var.ssh_key_path}")}"
+  }
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# CREATES A ROLE THAT IS ATTACHED TO THE INSTANCE
+# The arn of this AWS role is what the Vault server will use create the Vault Role
+# so it can validate login requests from resources with this role
+# ---------------------------------------------------------------------------------------------------------------------
+resource "aws_iam_instance_profile" "example_instance_profile" {
+  path = "/"
+  role = "${aws_iam_role.example_instance_role.name}"
+}
+
+resource "aws_iam_role" "example_instance_role" {
+  name_prefix        = "${var.auth_server_name}-role"
+  assume_role_policy = "${data.aws_iam_policy_document.example_instance_role.json}"
+}
+
+data "aws_iam_policy_document" "example_instance_role" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+  }
+}
+
+# Adds policies necessary for running consul
+module "consul_iam_policies_for_client" {
+  source = "github.com/hashicorp/terraform-aws-consul.git//modules/consul-iam-policies?ref=v0.3.10"
+
+  iam_role_id = "${aws_iam_role.example_instance_role.id}"
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# THE USER DATA SCRIPT THAT WILL RUN ON THE INSTANCE
+# This script will run consul, which is used for discovering vault cluster
+# And perform the login operation
+# ---------------------------------------------------------------------------------------------------------------------
+
+data "template_file" "user_data_auth_client" {
+  template = "${file("${path.module}/user-data-auth-client.sh")}"
+
+  vars {
+    consul_cluster_tag_key   = "${var.consul_cluster_tag_key}"
+    consul_cluster_tag_value = "${var.consul_cluster_name}"
+    example_role_name        = "${var.example_role_name}"
+  }
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# ADDS A RULE TO OPEN PORT 8080 SINCE OUR EXAMPLE LAUNCHES A SIMPLE WEB SERVER
+# ---------------------------------------------------------------------------------------------------------------------
```

Comment:
Clarify that you're only doing this for automated tests and wouldn't normally do this in a prod environment.